### PR TITLE
Classification of props file

### LINKED ISSUE #2852: Classify props file as XML
In VS land `.props` file are XML format, which contain the project properties and used by msbuild:
https://msdn.microsoft.com/en-us/library/669zx6zc.aspx

Example of such a file is https://github.com/dotnet/corefx/blob/67a17b3/dir.props

Please classify `.props` and `.xml`, so GitHub syntax highlighting can catch up.

cc @mellinoe, @rainersigwald

### PR DESCRIPTION
**Pull request by jasonwilliams200OK**
* Initially treat as XML
* Disambiguate from ini (key-value pair style with `=`)
* If the file is neither XML-style nor INI, classify as SQL

Fix #2852

### CODE REVIEW

File: lib/linguist/heuristics.rb [hunk 1]
```diff
@@ -313,6 +313,16 @@ def call(data)
       end
     end
 
+    disambiguate ".props" do |data|
+      if /^(\s*)(<Project|<Import|<Property|<\/Project|<\/Import|<\/Property)/i.match(data) 
+        Language["XML"]
+      elsif /\w+\s*=\s*/i.match(data)
+        Language["INI"]
+      else
```
Comment (pchaigno) [inline lib/linguist/heuristics.rb]: This is a bit dangerous and we try to avoid it. Isn't there a regular expression we could use here for SQL?

**Comment by pchaigno** [comment]
@jasonwilliams200OK Thanks for the pull request! I've added a comment above. You'll also need to add `.props` to SQL and INI for this to work properly ;)

File: lib/linguist/heuristics.rb (hunk 1 above)
Comment (jasonwilliams200OK) [inline lib/linguist/heuristics.rb]: Hi, is there a suitable language with this kind of syntax:

```
key1 value1
key2     value2
```
(this is ini/config, but without `=`)

sure I will add a regex for it. :)

**Comment by jasonwilliams200OK** [comment]
Thanks @pchaigno, if there any other suitable language which matches the pattern, I would add a regex and specify that lang.

```sql
# just used sql for this because I couldn't find a better match :(
key1 value1
key2   value2
```

**Comment by jasonwilliams200OK** [comment]
Analogically with INI specifier and `=` sign, the syntax highlighting works:

```ini
key1 =  value1
key2=value2
key3      = value3
```

But some `.props` files (configuration type) lack `=`.

File: lib/linguist/heuristics.rb (hunk 1 above)
Comment (jasonwilliams200OK) [inline lib/linguist/heuristics.rb]: Or should I just change the else case to `Text` until we found a better substitute?

**Comment by jasonwilliams200OK** [comment]
@arfon could you review this item please?

File: lib/linguist/heuristics.rb [hunk 2]
```diff
@@ -313,6 +313,16 @@ def call(data)
       end
     end
 
+    disambiguate ".props" do |data|
+      if /^(\s*)(<Project|<Import|<Property|<\/Project|<\/Import|<\/Property)/i.match(data)
```
Comment (arfon) [inline lib/linguist/heuristics.rb]: Could we simplify this to simply check for a `xmlns=`?

Comment (jasonwilliams200OK) [inline lib/linguist/heuristics.rb]: @arfon, xmlns is optional as there are over 3k files: https://github.com/search?p=2&q=extension%3Aprops+xml+version+encoding+OR+project+NOT+xmlns&ref=searchresults&type=Code&utf8=%E2%9C%93. However, I think we should add `<?xml` and `xmlns` checks and remove the closing tags patterns. Does it sound good?

Comment (arfon) [inline lib/linguist/heuristics.rb]: Yeah that sounds reasonable.

Comment (jasonwilliams200OK) [inline lib/linguist/heuristics.rb]: Done.

**Comment by jasonwilliams200OK** [comment]
@arfon, @pchaigno, I have removed the last case (https://github.com/github/linguist/pull/2866#issuecomment-193003371), which will be classified as text for now. Later when someone will know what is the closest language matching key-value-without-equals-sign, they can add it.

**Comment by arfon** [comment]
:+1: thanks @jasonwilliams200OK. This will be live in the next release of Linguist.

**Comment by jasonwilliams200OK** [comment]
Thanks @arfon! :tada:

**Pull request merged**